arclightning_backend/src/main.rs

```diff
@@ -0,0 +1,211 @@
+extern crate futures;
+extern crate hyper;
+#[macro_use]
+extern crate serde_derive;
+#[macro_use]
+extern crate serde_json;
+extern crate toml;
+
+use futures::future;
+use hyper::rt::{Future, Stream};
+use hyper::service::service_fn;
+use hyper::{Body, Method, Request, Response, Server, StatusCode};
+use std::path::PathBuf;
+
+// after this concept is further understood, will switch to 'Either'
+type ResponseFuture = Box<Future<Item = Response<Body>, Error = hyper::Error> + Send>;
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+// using PartialEq for unit tests
+// Using clone in a unit test atm.  Might not be necessary
+struct Game {
+    name: String,
+    description: String,
+    genre: Vec<String>,
+    thumbnail_path: PathBuf,
+    exe_path: PathBuf,
+}
+
+fn router(request: Request<Body>) -> ResponseFuture {
+    let mut games_list: Vec<Game> = Vec::new();
```

Comment:
Since this isn't used anymore, we can remove it

Reply:
We actually will be using it.  I will move the test case situation to be this function once it works properly.